### ISSUE TITLE
Formatted @value becomes incorrect mailto hyperlink

### ISSUE THREAD

**Issue by cyotek** [mike-ward/Markdown-Edit#142]
Spotted a minor issue while updating a document.

If you include a `@user` text in a document, then `@user` appears in the HTML as expected.

However, if I wrap the the text in `**` to make it bold, i.e. `**@user**`, MDE  then converts this into a mailto tag with the HTML `<a href="mailto:**@hello*">**@hello*</a>*`. Not only is this the wrong behaviour, it's also mucked it up - note the final `*` is excluded from the text and href.

I tested italics too, and observed the same incorrect behaviour(s).

(When I say convert, I mean the HTML Preview and HTML Export)

This only happens when GitHub markdown is active.

**Comment by cyotek** [comment]
Forgot to mention, changing the `@` symbol to its entity variant `&commat;` works around the issue, e.g. `**&commat;user**`

**Comment by mike-ward** [comment]
Which markup engine are you using?

**Comment by mike-ward** [comment]
Just tried this. I don't get the maito link. It doesn't convert to bold either.

![image](https://cloud.githubusercontent.com/assets/567927/14580339/775ebae6-0398-11e6-8f7a-ac60af8630e6.png)

Sample text?

**Comment by cyotek** [comment]
I spy with my little eye... a hyperlink in your screenshot :)

The MD content is fine. Your HTML preview rather clearly shows the hyperlink, and if you `Ctrl+E` to export the HTML, you'll see the mailto ref.

Using CommonMark, this issue doesn't happen. Enabling GitHub markdown triggers the issue.

**Comment by mike-ward** [comment]
Ha, I didn't see it. Subtle. This is a  pandoc issue. I'll report it there.

**Comment by mike-ward** [comment]
Submitted issue to pandoc (2940)

**Comment by mike-ward** [comment]
See @jgm response on pandoc 2940. Looks like we're stuck with this behavior.

**Comment by cyotek** [comment]
Hmm. That's a little annoying. The chance of actually anyone having an address which is ** or __ (or similar) is probably so incredibly tiny that I'd err on the other side of the fence and not translate it. Still, it's not my project and I can't dictate to others how they should do it so I'll just ignore the issue. As long as CommonMark/MarkdownDeep doesn't start doing something equally stupid then I'm happy. (And besides, there's the raw entity workaround if it bugs me overly much)

Thanks for looking into it anyway, sorry to keep dropping obscure problems on you that turn out to be from a third party each time :)

**Comment by mike-ward** [comment]
I have to agree. We know it's Markdown so let's be reasonable in our interpretation of an email address.

There's a brand new Markdown Engine in .NET that looks promising. Will be looking into using it for GitHub over Pandoc this weekend.

https://github.com/lunet-io/markdig

**Comment by cyotek** [comment]
Hmm that does look nice, especially the ++ and ~~, always grinds my goat having to write <ins> and <del> in Markdown documents. I think I shall check this one out myself too :)

**Comment by jgm** [comment]
+++ Mike Ward [May 27 16 05:57 ]:
>   See [1]@jgm response on pandoc 2940. Looks like we're stuck with this
>   behavior.

Hm, you all seem to have interpreted me as saying that I
like the current behavior and don't intend to change it.
I didn't say that, did I?

**Comment by mike-ward** [comment]
@jgm Apologies if we misunderstood. For the record, I didn't think you liked the behavior. :smile:
